### PR TITLE
Working basic MongoDB Tag DB

### PR DESCRIPTION
**Pull request by follen99**
This is a basic use of mongodb implementing CRUD operations on Tag class:
- addTag()
- getAllTags()
- deleteTagById()
- deleteTags()
- getTagsById()

### CODE REVIEW

File: qualitag/build.gradle [hunk 1]
```diff
@@ -3,6 +3,8 @@ plugins {
 	id 'org.springframework.boot' version '3.3.5'
 	id 'io.spring.dependency-management' version '1.1.6'
 	id 'com.vaadin' version '24.5.4'
+	// per installare automaticamente node
```
Comment (Mattia-Marino) [inline qualitag/build.gradle]: Comment in English, please

File: qualitag/build.gradle [hunk 2]
```diff
@@ -14,6 +16,17 @@ java {
 	}
 }
 
+node {
+	version = '16.13.0' // Specifica la versione di Node.js da installare
```
Comment (RaffaeleIzzo01) [inline qualitag/build.gradle]: Comments in english please

File: qualitag/build.gradle [hunk 3]
```diff
@@ -25,9 +38,13 @@ ext {
 dependencies {
 	implementation 'org.springframework.boot:spring-boot-starter-web'
 	implementation 'com.vaadin:vaadin-spring-boot-starter'
+	implementation 'org.projectlombok:lombok'
```
Comment (Mattia-Marino) [inline qualitag/build.gradle]: Please remove Lombok, at least for now

File: qualitag/src/main/resources/application.properties [hunk 4]
```diff
@@ -1,2 +1,20 @@
...
+#spring.datasource.driver-class-name=org.h2.Driver
+#spring.datasource.url=jdbc:h2:mem:qualitag
+##spring.datasource.url=jdbc:h2:file:./src/main/resources/qualitag
+##spring.jpa.hibernate.ddl-auto=update
+#spring.datasource.username=test
+#spring.datasource.password=password
+#spring.sql.init.schema-locations=classpath:schema.sql
+
+# NEW STUFF, USING MONGODB DATABASE
+#//mongodb://localhost:27017
```
Comment (Mattia-Marino) [inline qualitag/src/main/resources/application.properties:6]: Remove old stuff

File: qualitag/src/main/resources/schema.sql
Comment (Mattia-Marino) [inline qualitag/src/main/resources/schema.sql:1]: Useless now with MongoDB

Comment (RaffaeleIzzo01) [inline qualitag/src/main/resources/schema.sql:1]: Useless now that we decided to use MongoDB

File: qualitag/src/main/java/it/unisannio/studenti/qualitag/model/Artifact.java [hunk 5]
```diff
@@ -0,0 +1,162 @@
+package it.unisannio.studenti.qualitag.model;
+
+import java.util.ArrayList;
+import java.util.Objects;
+
+public class Artifact {
+
+  private String artifact_id;
```
Comment (RaffaeleIzzo01) [inline qualitag/src/main/java/it/unisannio/studenti/qualitag/model/Artifact.java:8]: Undescores create problems with MongoDB

File: qualitag/src/main/java/it/unisannio/studenti/qualitag/model/Project.java [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+ * Created by Raffaele Izzo on 14/11/2024
+ */
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Objects;
+
+public class Project {
+
+  private String project_id;
```
Comment (RaffaeleIzzo01) [inline qualitag/src/main/java/it/unisannio/studenti/qualitag/model/Project.java:15]: Other underscores here

File: qualitag/src/main/java/it/unisannio/studenti/qualitag/model/Team.java [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Represents a team in the system.
+ */
+public class Team {
+
+  private String team_id;
+  private String project_id;
```
Comment (RaffaeleIzzo01) [inline qualitag/src/main/java/it/unisannio/studenti/qualitag/model/Team.java:14]: other underscores here

**Review by Mattia-Marino — CHANGES_REQUESTED** [review]
Great improvements but need some polishing

File: qualitag/src/test/java/it/unisannio/studenti/qualitag/model/ArtifactTest.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ArtifactTest {
+
+    @Test
+    public void testArtifactDefaultConstructor() {
+        Artifact artifact = new Artifact();
+        artifact.setArtifact_id("default_id");
```
Comment (RaffaeleIzzo01) [inline qualitag/src/test/java/it/unisannio/studenti/qualitag/model/ArtifactTest.java:12]: Underscores

File: qualitag/src/test/java/it/unisannio/studenti/qualitag/model/ProjectTest.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ProjectTest {
+
+    @Test
+    public void testProjectDefaultConstructor() {
+        Project project = new Project();
+
+        project.setProject_id("default_id");
```
Comment (RaffaeleIzzo01) [inline qualitag/src/test/java/it/unisannio/studenti/qualitag/model/ProjectTest.java:15]: Underscores

File: qualitag/src/test/java/it/unisannio/studenti/qualitag/model/TeamTest.java [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+import java.util.Date;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class TeamTest {
+    @Test
+    public void testTeamConstructor() {
+        Team team = new Team("1", "1", "Test", null, "Test");
+        assertEquals("1", team.getTeam_id());
```
Comment (RaffaeleIzzo01) [inline qualitag/src/test/java/it/unisannio/studenti/qualitag/model/TeamTest.java:13]: Underscores

**Review by RaffaeleIzzo01 — CHANGES_REQUESTED** [review]
Needs a few changes, it's mostly fine

**Review by Mattia-Marino — APPROVED** [review]
All good now

**Review by RaffaeleIzzo01 — APPROVED** [review]

**Pull request merged**